client/web/src/backend/graphql.ts

```diff
@@ -4,12 +4,18 @@ import { Observable } from 'rxjs'
 import { getGraphQLClient, GraphQLResult, requestGraphQLCommon } from '@sourcegraph/http-client'
 import * as GQL from '@sourcegraph/shared/src/schema'
 
-const getHeaders = (): { [header: string]: string } => ({
-    ...window?.context?.xhrHeaders,
-    Accept: 'application/json',
-    'Content-Type': 'application/json',
-    'X-Sourcegraph-Should-Trace': new URLSearchParams(window.location.search).get('trace') || 'false',
-})
+const getHeaders = (): Headers => {
+    const headers = new Headers({
+        ...window?.context?.xhrHeaders,
```

Comment:
This code does not look OK to me. 

`window` should always be defined in a browser context. Do we ever run this code outside of browser, e.g. E2E tests? 

I also think the code would be more readable if we provide a default here, e.g.:
```suggestion
        ...window?.context?.xhrHeaders || {},
```

Reply:
Look at the old code, I'm just copying what was done before. I have no idea if it is ok :) I was just confused why switching from the `{ [header: string]: string }` type to `Headers` caused the breakage. (I see your explanation of node v16 vs v18, and that makes sense I suppose. I don't know enough about our frontend stack to comment otherwise)